analyzers/src/SonarAnalyzer.CSharp/Facade/CSharpSyntaxFacade.cs

```diff
@@ -149,4 +150,7 @@ node switch
 
     public override bool IsInExpressionTree(SemanticModel model, SyntaxNode node) =>
         node.IsInExpressionTree(model);
+
+    public override bool IsWrittenTo(SyntaxNode expression, SemanticModel semanticModel, CancellationToken cancellationToken) =>
+        expression is ExpressionSyntax ex && ex.IsWrittenTo(semanticModel, cancellationToken);
```

Comment:
Same as the comment for VB
```suggestion
        expression is ExpressionSyntax { } ex && ex.IsWrittenTo(semanticModel, cancellationToken);
```